### PR TITLE
Fix parsing of invalid asterix symbol after private field symbol

### LINKED ISSUE #5153: SEGV in parser_parse_class and parser_parse_for_statement_start
###### Commit ID
2dbb6f79efa08767ce44fbc8e3f26e80649295d7

###### Build platform
Ubuntu 22.04.3

###### Build steps
```sh
python3 ./tools/build.py --builddir=xxx --clean --compile-flag=-fsanitize=address --compile-flag=-g --strip=off --lto=off --logging=on --line-info=on --error-message=on --stack-limit=20
```

###### Execution steps
```sh
./jerry poc.js
```

###### Test case
```sh
//poc1.js
class C { async#* method () {
    class 
```

###### Output
```sh
AddressSanitizer:DEADLYSIGNAL
=================================================================
==2742590==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000008 (pc 0x0000007a8144 bp 0x7fff473ce8f0 sp 0x7fff473ce6a0 T0)
==2742590==The signal is caused by a READ memory access.
==2742590==Hint: address points to the zero page.
    #0 0x7a8144 in parser_parse_class /jerryscript/jerry-core/parser/js/js-parser-expr.c:1016:46
    #1 0x7dabc6 in parser_parse_statements /jerryscript/jerry-core/parser/js/js-parser-statm.c:2787:9
    #2 0x647e67 in parser_parse_function /jerryscript/jerry-core/parser/js/js-parser.c:2717:3
    #3 0x79eec2 in lexer_construct_function_object /jerryscript/jerry-core/parser/js/js-lexer.c:2695:23
    #4 0x7aa79e in parser_parse_class_body /jerryscript/jerry-core/parser/js/js-parser-expr.c:911:39
    #5 0x7aa79e in parser_parse_class /jerryscript/jerry-core/parser/js/js-parser-expr.c:1113:27
    #6 0x7dabc6 in parser_parse_statements /jerryscript/jerry-core/parser/js/js-parser-statm.c:2787:9
    #7 0x663433 in parser_parse_source /jerryscript/jerry-core/parser/js/js-parser.c:2280:5
    #8 0x6611ca in parser_parse_script /jerryscript/jerry-core/parser/js/js-parser.c:3332:38
    #9 0x4ec611 in jerry_parse_common /jerryscript/jerry-core/api/jerryscript.c:418:21
    #10 0x4ec23e in jerry_parse /jerryscript/jerry-core/api/jerryscript.c:486:10
    #11 0x80402d in jerryx_source_parse_script /jerryscript/jerry-ext/util/sources.c:52:26
    #12 0x80432c in jerryx_source_exec_script /jerryscript/jerry-ext/util/sources.c:63:26
    #13 0x4e0160 in main /jerryscript/jerry-main/main-desktop.c:156:20
    #14 0x7fdcb820ad8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
    #15 0x7fdcb820ae3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f)
    #16 0x427fd4 in _start (/jerryscript/0805/bin/jerry+0x427fd4)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /jerryscript/jerry-core/parser/js/js-parser-expr.c:1016:46 in parser_parse_class
==2742590==ABORTING
```

###### Test case
```sh
//poc2.js
class C { async#* method () {
    for ()
```

###### Output
```sh
AddressSanitizer:DEADLYSIGNAL
=================================================================
==2736108==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000008 (pc 0x0000007dae99 bp 0x7ffe22722210 sp 0x7ffe22721760 T0)
==2736108==The signal is caused by a READ memory access.
==2736108==Hint: address points to the zero page.
    #0 0x7dae99 in parser_parse_for_statement_start /jerryscript/jerry-core/parser/js/js-parser-statm.c:1137:62
    #1 0x7dae99 in parser_parse_statements /jerryscript/jerry-core/parser/js/js-parser-statm.c:2851:9
    #2 0x647e67 in parser_parse_function /jerryscript/jerry-core/parser/js/js-parser.c:2717:3
    #3 0x79eec2 in lexer_construct_function_object /jerryscript/jerry-core/parser/js/js-lexer.c:2695:23
    #4 0x7aa79e in parser_parse_class_body /jerryscript/jerry-core/parser/js/js-parser-expr.c:911:39
    #5 0x7aa79e in parser_parse_class /jerryscript/jerry-core/parser/js/js-parser-expr.c:1113:27
    #6 0x7dabc6 in parser_parse_statements /jerryscript/jerry-core/parser/js/js-parser-statm.c:2787:9
    #7 0x663433 in parser_parse_source /jerryscript/jerry-core/parser/js/js-parser.c:2280:5
    #8 0x6611ca in parser_parse_script /jerryscript/jerry-core/parser/js/js-parser.c:3332:38
    #9 0x4ec611 in jerry_parse_common /jerryscript/jerry-core/api/jerryscript.c:418:21
    #10 0x4ec23e in jerry_parse /jerryscript/jerry-core/api/jerryscript.c:486:10
    #11 0x80402d in jerryx_source_parse_script /jerryscript/jerry-ext/util/sources.c:52:26
    #12 0x80432c in jerryx_source_exec_script /jerryscript/jerry-ext/util/sources.c:63:26
    #13 0x4e0160 in main /jerryscript/jerry-main/main-desktop.c:156:20
    #14 0x7f064248dd8f  (/lib/x86_64-linux-gnu/libc.so.6+0x29d8f)
    #15 0x7f064248de3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e3f)
    #16 0x427fd4 in _start (/jerryscript/0805/bin/jerry+0x427fd4)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /jerryscript/jerry-core/parser/js/js-parser-statm.c:1137:62 in parser_parse_for_statement_start
==2736108==ABORTING
```

### PR DESCRIPTION
**Pull request by kulcsaradam**
Fixes #5153

### CODE REVIEW

File: jerry-core/parser/js/js-parser-expr.c [hunk 1]
```diff
@@ -687,6 +687,11 @@ parser_parse_class_body (parser_context_t *context_p, /**< context */
         context_p->token.flags |= LEXER_NO_SKIP_SPACES;
         ident_opts |= LEXER_OBJ_IDENT_CLASS_PRIVATE;
         is_private = true;
+
+        if (lexer_check_next_character (context_p, LIT_CHAR_ASTERISK))
+        {
+          parser_raise_error (context_p, PARSER_ERR_INVALID_CHARACTER);
+        }
```
Comment (zherczeg) [inline jerry-core/parser/js/js-parser-expr.c:694]: There is a `if (is_private && context_p->token.type == LEXER_LITERAL)` below. Would it be a good idea to move it here?

Comment (kulcsaradam) [reply jerry-core/parser/js/js-parser-expr.c:694]: Moving this condition ruins the some jerry and 262 tests. Mainly those with private field names.

**Comment by kulcsaradam** [comment]
Updated the commit message according to the CI.

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by akosthekiss — APPROVED** [review]
lgtm

**Pull request merged**